src/features/bank/Bank.tsx

```diff
@@ -35,23 +35,28 @@ export const Bank: React.FC = () => {
         top: `${GRID_WIDTH_PX * 4.65}px`,
       }}
     >
-      <img
-        src={bank}
-        alt="bank"
-        onClick={isNotReadOnly ? open : undefined}
-        className={classNames("w-full", {
+      <div
+        className={classNames({
           "cursor-pointer": isNotReadOnly,
           "hover:img-highlight": isNotReadOnly,
         })}
-      />
-      {isNotReadOnly && (
-        <Action
-          className="absolute -bottom-6 left-2"
-          text="Bank"
-          icon={token}
-          onClick={open}
+      >
+        <img
+          src={bank}
+          alt="bank"
+          onClick={isNotReadOnly ? openBank : undefined}
```

Comment:
You can move this check inside `openBank` function and if `isNotReadOnly` you just return before executing anything.

Reply:
Moved to openBank function. Thanks! 😄